typings/index.d.ts

```diff
@@ -3138,23 +3213,48 @@ declare module 'discord.js' {
 
   type MessageAdditions = MessageEmbed | MessageAttachment | (MessageEmbed | MessageAttachment)[];
 
+  interface MessageActionRowOptions extends BaseMessageComponentOptions {
+    type: 'ACTION_ROW';
+    components?: MessageComponentResolvable[];
+  }
+
   interface MessageActivity {
     partyID: string;
     type: number;
   }
 
+  interface MessageButtonOptions extends BaseMessageComponentOptions {
+    customID?: string;
```

Comment:
Why are all of these optional? Shouldn't at least `customID` and `style` be required?

Reply:
`style` is the only one that can be enforced. `customID` isn't required on a `LINK` style button.